### PR TITLE
Adding in instructions for how to install Kustomize 3.2.1 on a Linux box

### PR DESCRIPTION
**Pull request by jmdevcs**
Just wanted to add quick instructions for how to install Kustomize 3.2.1 on a Linux box.

### CODE REVIEW

**Comment by jmdevcs** [comment]
@googlebot I signed it!

File: content/en/docs/gke/deploy/deploy-cli.md
Comment (8bitmp3) [inline content/en/docs/gke/deploy/deploy-cli.md:53]: Suggestion: 

1. Use `run the following commands` instead of `follow these commands`, since we're technically typing them in and then running them. (We, as users, do follow steps though.)

```diff
-    To deploy Kustomize v3.2.1 on a Linux box, follow these commands:
+    To deploy Kustomize v3.2.1 on a Linux box, follow these commands:
...
```

2. Provide a bit more information on how the user should know if the installation went OK. For example:

```diff
-     Then run `kustomize version` to verify the install.
+     Then, to verify the installation, run `kustomize version`. You should see the version of Kustomize in the output if you've successfully deployed it.
```

(while keeping each line at 80 characters max)

WDYT?

**Review by 8bitmp3 — CHANGES_REQUESTED** [review]
Hey @jmdevcs Thank you for the PR!

A few suggestions from my side. 

1. Use `run the following commands` instead of `follow these commands`, since we're technically typing them in and then running them. (We, as users, do follow steps though.)

```diff
-    To deploy Kustomize v3.2.1 on a Linux box, follow these commands:
+    To deploy Kustomize v3.2.1 on a Linux box, run the following commands:
...
```

2. Provide a bit more information on how the user should know if the installation went OK. For example:

```diff
-     Then run `kustomize version` to verify the install.
+     Then, to verify the installation, run `kustomize version`. You should see the version of Kustomize in the output if you've successfully deployed it.
```

(while keeping each line at 80 characters max)

WDYT?

Do double check with the folks who are good at Kustomize stuff that they're OK with your added content.

Thanks! 👏

**Comment by 8bitmp3** [comment]
@jmdevcs Can you pls also investigate the build log file and see why the test is failing?

File: content/en/docs/gke/deploy/deploy-cli.md
Comment (jmdevcs) [reply content/en/docs/gke/deploy/deploy-cli.md:53]: I agree, the wording is better! Thanks!

**Comment by jmdevcs** [comment]
@8bitmp3 Looks like the run failed due to the test suite being unable to connect to `ouath2.googleapis.com`. This would be out of my control.

https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubeflow_website/2169/kubeflow-website-presubmit/1300895492493807616/#1:build-log.txt%3A59

```
ERROR|2020-09-01T20:37:45|/src/kubeflow/testing/py/kubeflow/testing/run_e2e_workflow.py|487| Other exception: HTTPSConnectionPool(host='oauth2.googleapis.com', port=443): Max retries exceeded with url: /token (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f1eca377950>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',))
```

**Comment by jmdevcs** [comment]
@8bitmp3 Looks like the test is passing now.

**Comment by 8bitmp3** [comment]
@jmdevcs Awesome. Please go ahead and make the changes and we can re-test and move forward. Thanks!

**Comment by jmdevcs** [comment]
@8bitmp3 Done, text updated. Thanks!

**Comment by jmdevcs** [comment]
@rmgogogo @8bitmp3 Would someone be able to review and approve this PR, if possible?

Thank you!

**Comment by jmdevcs** [comment]
@8bitmp3 What are the next steps needed ?

**Comment by 8bitmp3** [comment]
Hey @jmdevcs The approvers of `gke` docs are @Bobgy @joeliedtke and @rmgogogo  (https://github.com/kubeflow/website/blob/master/content/en/docs/gke/OWNERS). Let's do another `/assign` since you've already assigned @rmgogogo :

/assign @joeliedtke @Bobgy

**Comment by 8bitmp3** [comment]
@jmdevcs Let's see what the GCP team says.

Also, this may sound better - we can ask for their opinion:

```diff
- Then, to verify the installation, run `kustomize version`. You should see `Version:kustomize/v3.2.1` in the output if you've successfully deployed it.
+ Then, to verify the installation, run `kustomize version`. You should see `Version:kustomize/v3.2.1` in the output if you've successfully deployed Kustomize.
```

**Comment by Bobgy** [comment]
Thank you for the PR! And thanks @8bitmp3 for the suggestion!
Sorry coming late here
/lgtm

I'll need @joeliedtke for a final tech writer review.

**Comment by joeliedtke** [comment]
Preview:
https://deploy-preview-2169--competent-brattain-de2d6d.netlify.app/docs/gke/deploy/deploy-cli/

File: content/en/docs/gke/deploy/deploy-cli.md
Comment (joeliedtke) [inline content/en/docs/gke/deploy/deploy-cli.md:59]: Consider removing this line, adding a comment to indicate that it is optional, or adding it as an optional step. Users may need to have different versions of Kustomize (since we are asking them to use something other than current version), and adding moving Kustomize into the search path may cause unexpected behaviors.

**Review by joeliedtke — CHANGES_REQUESTED** [review]
Thanks for adding these instructions. I've added a couple suggestions, but otherwise this looks great to me.

File: content/en/docs/gke/deploy/deploy-cli.md
Comment (jmdevcs) [reply content/en/docs/gke/deploy/deploy-cli.md]: Done! Changed to "machine".

File: content/en/docs/gke/deploy/deploy-cli.md
Comment (jmdevcs) [reply content/en/docs/gke/deploy/deploy-cli.md:60]: Well, the docs say that Kustomize v3.2.1 is recommended. However, I see your point -- users may have earlier versions of Kustomize or their own Kustomize setup altogether.

I've added the following text:

` # These steps are optional`

**Comment by jmdevcs** [comment]
@8bitmp3 @joeliedtke Changes made! Thank you! :)

**Comment by 8bitmp3** [comment]
@joeliedtke @Bobgy @rmgogogo Let us know if this looks OK. If it is 👍 please `/approve` Thank you!

**Comment by rmgogogo** [comment]
Wondering why not encourage to use "kubectl -k" to avoid install kustomize. My knowledge is that "kubectl -k" and "kustomize" still has few compatibility/gap items.

@Bobgy for giving approval. I'm OK with this PR or revert to use "kubectl -k".

**Comment by jmdevcs** [comment]
> Wondering why not encourage to use "kubectl -k" to avoid install kustomize. My knowledge is that "kubectl -k" and "kustomize" still has few compatibility/gap items.
> 
> @Bobgy for giving approval. I'm OK with this PR or revert to use "kubectl -k".

I don't believe `kubectl -k` is a an actual command. You may be referring to `kubectl kustomize` which just lists out the K8 objects created by Kustomize, rather than install Kustomize itself. A lot of the GCP manifest files use the kustomize api & kustomize-specific features.

File: content/en/docs/gke/deploy/deploy-cli.md
Comment (Bobgy) [reply content/en/docs/gke/deploy/deploy-cli.md:60]: Hi @jmdevcs and @joeliedtke, I think I need to clarify sth here:
* Kustomize v3.2.1 is not optional for deploying Kubeflow 1.1. The version range that supports Kubeflow is very strict, if you use kustomize v2.x or early v3.x some new features required by Kubeflow haven't been there. Also, if you use Kustomize v3.3+, a breaking change prevents everything from working.
* I believe what @joeliedtke meant was that `sudo mv ./kustomize /usr/local/bin/kustomize` this single line is optional, which I believe is not true either. In Kubeflow on GCP deploy scripts, it calls `kustomize` command, so we must make sure `kustomize` command is in `$PATH`.

My suggestions:
1. Remove the optional statement
2. Add a comment before `sudo mv ./kustomize /usr/local/bin/kustomize`, saying we need to add the `kustomize` tool in your `$PATH`, but you may choose other ways to do so (e.g. change $PATH variable).

**Comment by Bobgy** [comment]
@rmgogogo for clarification, this doc is about deploying Kubeflow and Kustomize ~3.2.1 is required. It seems you confused it with the documentation to deploy Kubeflow Pipelines Standalone.

More context in https://github.com/kubeflow/website/pull/2169#discussion_r491778991

**Comment by rmgogogo** [comment]
> @rmgogogo for clarification, this doc is about deploying Kubeflow and Kustomize ~3.2.1 is required. It seems you confused it with the documentation to deploy Kubeflow Pipelines Standalone.
> 
> More context in [#2169 (comment)](https://github.com/kubeflow/website/pull/2169#discussion_r491778991)

Thanks. I didn't notice KF full set requires Kustomize (the part "kubectl -k" can't work). Feel free to approve. I'm OK with this PR.

File: content/en/docs/gke/deploy/deploy-cli.md
Comment (jmdevcs) [reply content/en/docs/gke/deploy/deploy-cli.md:61]: Hi @Bobgy 

Thank you for the clarification! That was my initial understanding as well.

I have made the recommended suggestions @Bobgy

**Comment by jmdevcs** [comment]
Requested updates made @Bobgy

**Comment by Bobgy** [comment]
Thank you @jmdevcs !
/lgtm
/approve

**Pull request merged**
